src/linter/utils.go

```diff
@@ -90,6 +90,32 @@ func typesMapToTypeExpr(p *phpdoc.TypeParser, m meta.TypesMap) phpdoc.Type {
 	return p.Parse(typeString)
 }
 
+// functionReturnType returns the return type of a function over computed types
+// according to the convention below:
+//
+// The types are inferred as follows:
+// 1. If there is an @param annotation, then its value becomes the return type;
+//
+// 2. If there is a type hint, then it is added to the types from the @param.
+//    If the @param is empty, then the type matches the type hint itself;
```

Comment:
Does merging type hint and `@return` improve the type inference?

For example, in new tests there is `f6()` which has `array` type hint and `int[]` from phpdoc annotation.
I would guess that marking `f6()` result as `int[]` is more desirable than `int[]|mixed[]`.

Reply:
Done